tests/x11/vnc_two_passwords.pm

```diff
@@ -38,47 +39,67 @@ my @options = ({pw => "readonly_pw", change => 0}, {pw => "readwrite_pw", change
-    assert_script_run('chmod go-rwx "$HOME/.vnc"');
-    if (script_run("echo \"$options[1]->{pw}\" | vncpasswd -f > \$HOME/.vnc/passwd; echo \"$options[0]->{pw}\" | vncpasswd -f >> \$HOME/.vnc/passwd") != 0) {
-        record_soft_failure('vncpasswd crashes - bsc#1171519');
-        assert_script_run('cp /tmp/file.passwd $HOME/.vnc/passwd');
+    if ($curr_vers < 0) {
+        assert_script_run('umask 0077');
+        assert_script_run('mkdir -Z $HOME/.vnc');
+        assert_script_run('chmod go-rwx "$HOME/.vnc"');
+        script_output(qq(expect -c 'spawn vncpasswd /tmp/file.passwd; expect "Password:"; send $options[1]->{pw}\\r; expect "Verify:"; send $options[1]->{pw}\\r; expect "view-only"; send y\\r; expect "Password:"; send $options[0]->{pw}\\r; expect "Verify:"; send $options[0]->{pw}\\r ; interact'));
+        if (script_run("echo \"$options[1]->{pw}\" | vncpasswd -f > \$HOME/.vnc/passwd; echo \"$options[0]->{pw}\" | vncpasswd -f >> \$HOME/.vnc/passwd") != 0) {
+            record_soft_failure('vncpasswd crashes - bsc#1171519');
+            assert_script_run('cp /tmp/file.passwd $HOME/.vnc/passwd');
+        }
+        wait_still_screen 2;
+        record_info("TigerVNC version", "TigerVNC version $tigervnc_vers is lesser than 1.12.0");
+        assert_script_run("vncserver $display -geometry 1024x768 -depth 16", fail_message => "vncserver is not starting");
+        script_run("vncserver -list > /var/tmp/vncserver-list");
+        assert_script_run("grep '$display' /var/tmp/vncserver-list", fail_message => "vncserver is not running");
+        set_serial_prompt '# ';
+        enter_cmd 'exit';
+        sleep 2;
+    }
+    else {
+        script_output(qq(expect -c 'spawn vncpasswd; expect "Password:"; send $options[1]->{pw}\\r; expect "Verify:"; send $options[1]->{pw}\\r; expect "view-only"; send y\\r; expect "Password:"; send $options[0]->{pw}\\r; expect "Verify:"; send $options[0]->{pw}\\r ; interact'));
+        script_output("echo \"$options[1]->{pw}\" | vncpasswd -f");    # bsc#1171519
+        script_output(qq(expect -c 'spawn vncpasswd -f /home/test/.config/tigervnc/passwd; expect "Password:"; send $options[1]->{pw}\\r; expect "Verify:"; send $options[1]->{pw}\\r; expect "view-only"; send y\\r; expect "Password:"; send $options[0]->{pw}\\r; expect "Verify:"; send $options[0]->{pw}\\r ; interact'));
+        record_info("TigerVNC version", "TigerVNC version $tigervnc_vers is greater than of equal to 1.12.0");
+        set_serial_prompt '# ';
+        enter_cmd 'exit';
+        sleep 2;
+        assert_script_run("systemctl start vncserver\@$display", fail_message => "New version of vncserver is not starting");
+        assert_script_run("systemctl status vncserver\@$display", fail_message => "New version of vncserver is not running");
+        sleep 2;
     }
+    # get xauth of test user so bernhard can access test display :37 with xev
+    $test_xauth = script_output("sudo -u test xauth list|awk '{print\$3}'|head -n1");
```

Comment:
Instead of `sudo -u` here, run it before `exit`?

Reply:
Alternatively you can also do `xhost +si:localuser:$testapi::username` to allow bernhard access to test's display

Reply:
I will not use xhost, probably easier, but as we not login testuser, there will be problem getting the DISPLAY `xhost:  unable to open display ""`
I don't want to spend any more unneeded time on this test, too much time spend here ...
The command must be executed after vnc is started when the DISPLAY does exist.